### ISSUE TITLE
mc: <ERROR> Unable to list service accounts: json: cannot unmarshal string into Go struct

### ISSUE THREAD

**Issue by rino-pupkin** [minio/mc#4589]
## Expected behavior
Show service accounts list for user
```
[root@9b14bd603ac4 ~]# mc admin user svcacct list testsrv/ user
AGVKR9O0N5PGP13PYE45

[root@9b14bd603ac4 ~]# mc --version
mc version RELEASE.2023-05-18T16-59-00Z (commit-id=43745a9a597547f6bcd381b63b2e3c509d9c51ac)
Runtime: go1.19.9 linux/amd64
Copyright (c) 2015-2023 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
```

## Actual behavior
Throws error. 
```
[root@bfefa66eca43 ~]# mc admin user svcacct list testsrv/ user
mc: <ERROR> Unable to list service accounts: json: cannot unmarshal string into Go struct field ListServiceAccountsResp.accounts of type madmin.ServiceAccountInfo.

[root@bfefa66eca43 ~]# mc admin user svcacct info testsrv/ AGVKR9O0N5PGP13PYE45
AccessKey: AGVKR9O0N5PGP13PYE45
ParentUser: user
Status: on
Name: appsvc
Description: 
Policy: implied
Expiration: no-expiry

[root@bfefa66eca43 ~]# mc version RELEASE.2023-05-26T23-31-54Z (commit-id=9cb069e7afaa45c64c45b7b59ba65c0441019efd)
Runtime: go1.19.9 linux/amd64
Copyright (c) 2015-2023 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
```

## Steps to reproduce the behavior
1. run mc latest 
2. create user
3. create and attach policy to user
4. create service account for user
5. try to list user's service accounts

## mc --version
mc version RELEASE.2023-05-26T23-31-54Z (commit-id=9cb069e7afaa45c64c45b7b59ba65c0441019efd)
Runtime: go1.19.9 linux/amd64
Copyright (c) 2015-2023 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>

## System information
Works well in RELEASE.2023-05-18T16-59-00Z
Doesn't work in RELEASE.2023-05-26T23-31-54Z

```
[root@bfefa66eca43 ~]# mc admin user svcacct list testsrv/ user --debug 
mc: <DEBUG> GET /minio/admin/v3/list-service-accounts?user=user HTTP/1.1
Host: 172.17.0.1:9000
User-Agent: MinIO (linux; amd64) madmin-go/2.0.0 mc/RELEASE.2023-05-26T23-31-54Z
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=s3root/20230602//s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20230602T082117Z

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 94
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/json
Date: Fri, 02 Jun 2023 08:21:17 GMT
Server: MinIO
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Id-2: dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8
X-Amz-Request-Id: 1764C9D51A30C567
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  45.237425ms

mc: <ERROR> Unable to list service accounts: json: cannot unmarshal string into Go struct field ListServiceAccountsResp.accounts of type madmin.ServiceAccountInfo
 (1) admin-user-svcacct-list.go:80 cmd.mainAdminUserSvcAcctList(..) Tags: [testsrv/, user]
 (0) admin-user-svcacct-list.go:80 cmd.mainAdminUserSvcAcctList(..)
 Release-Tag:RELEASE.2023-05-26T23-31-54Z | Commit:9cb069e7afaa | Host:bfefa66eca43 | OS:linux | Arch:amd64 | Lang:go1.19.9 | Mem:67 MiB/90 MiB | Heap:67 MiB/76 MiB.

```

**Comment by harshavardhana** [comment]
@rino-pupkin what is the server version here?

**Comment by rino-pupkin** [comment]
@harshavardhana 
tryed with servers 
```
RELEASE.2023-05-18T00-05-36Z
RELEASE.2023-05-27T05-56-19Z
```

**Comment by harshavardhana** [comment]
Looks like `mc` updated the marshaling requirement, the server has not yet. Please use RELEASE.2023-05-18T16-59-00Z for now.

**Comment by harshavardhana** [comment]
@kaankabalak assigning this to you so that we can prioritize the merge of the server-side PR.

**Comment by kaankabalak** [comment]
Should be fixed as minio/minio#17249 has been merged, after the new MinIO Server release you shouldn't be seeing this issue anymore.
